### ISSUE TITLE
Update has() macro specification to support repeated proto fields

### ISSUE THREAD

**Issue by TristonianJones** [google/cel-spec#113]
In proto2 all non-repeated fields are represented by pointers which will be nil if not set. In proto3 all message fields are represented by pointers which will be nil if not set. The has() macro must also return false if the proto3 primitive field is equal to its zero value.

It seems worth reexamining whether the semantics for proto3 tests of defaultness should also be extended to repeated types (array, map) for both proto2 and proto3 where:

```
has(e.repeated_f) returns false if repeated_f contains zero elements. 
```

Stylistically the following expressions would be equivalent and harmless:

```
// equivalent to e.map_f.size() == 0 || has(e.map_f["key"])
!has(e.map_f) || has(e.map_f["key"])

// equivalent to e.list_f.size() > 0 && e.list_f[0]
has(e.list_f) && e.list_f[0]
```

This change better aligns `has` between proto and json, especially for maps. The one key difference would be that for json lists, the has() would only indicate the definition of the field and would not indicate a non-empty value:

```
// Currently valid expression over a JSON object.
!has(json.map_f) || has(json.map_f["key"])

// Slight difference from the proto usage, but this is already a difference.
has(json.list_f) && json.list_f.size() > 0 && json.list_f[0]
```

**Comment by JimLarson** [comment]
I'd think of it more as a difference between protobuf messages and protobuf maps, with JSON data following the map behavior. Yes, I'm okay with the behavior clarification. It's compatible with the current semantics doc.

**Comment by JimLarson** [comment]
Closed by PR #115.
